thewalrus/tests/test_quantum.py

```diff
@@ -1854,3 +1855,28 @@ def test_single_mode_first_and_second_cumulant(hbar):
     obtained_var = click_cumulant(mu, cov, [0, 0], hbar=hbar)
     assert np.allclose(expected_mean, obtained_mean)
     assert np.allclose(expected_var, obtained_var)
+
+
+@pytest.mark.parametrize("N", range(2, 20, 2))
+def test_is_symplectic(N):
+    """Tests the is_symplectic function for different matrix sizes"""
+    S = random_symplectic(N)
+    assert is_symplectic(S)
+
+
+def test_is_symplectic_rect():
+    """Testing that rectangular matrices return False in is_symplectic"""
+    S = np.random.rand(2, 3)
+    assert not is_symplectic(S)
+
+
+def test_is_symplectic_odd():
+    """Testing that odd-sized matrices return False in is_symplectic"""
+    S = np.random.rand(3, 3)
+    assert not is_symplectic(S)
+
+
+def test_is_symplectic_false():
+    """Testing that non-symplectic matrices return False in is_symplectic"""
+    S = np.random.rand(4, 4)

```

Comment:
Even though I assume the possibility that `S` will turn out to be symplectic is very low, it might be better to hard-code a known non-symplectic matrix here instead (just to make sure that the test _always_ works).

Reply:
is not low, it is zero.

Reply:
Oh, is this because a symplectic matrix always will contain at least one entry that's either negative or >1?

Reply:
is because Symplectic matrices are very special. For example a necessary (but not so sufficient!) condition to be symplectic is to have unit determinant.